crates/holochain_sqlite/src/conn.rs

```diff
@@ -0,0 +1,140 @@
+use crate::prelude::*;
+use chashmap::CHashMap;
+use lazy_static::lazy_static;
+use rusqlite::*;
+use std::{
+    path::{Path, PathBuf},
+    time::Duration,
+};
+
+mod singleton_conn;
+
+const SQLITE_BUSY_TIMEOUT: Duration = Duration::from_secs(30);
+
+lazy_static! {
+
+    pub(crate) static ref DATABASE_HANDLES: CHashMap<PathBuf, DbWrite> = {
+        // This is just a convenient place that we know gets initialized
+        // both in the final binary holochain && in all relevant tests
+        //
+        // Holochain (and most binaries) are left in invalid states
+        // if a thread panic!s - switch to failing fast in that case.
+        //
+        // We tried putting `panic = "abort"` in the Cargo.toml,
+        // but somehow that breaks the wasmer / test_utils integration.
+
+        let orig_handler = std::panic::take_hook();
+        std::panic::set_hook(Box::new(move |panic_info| {
+            // print the panic message
+            eprintln!("FATAL PANIC {:#?}", panic_info);
+            // invoke the original handler
+            orig_handler(panic_info);
+            // // Abort the process
+            // // TODO - we need a better solution than this, but if there is
+            // // no better solution, we can uncomment the following line:
+            // std::process::abort();
```

Comment:
I don't follow... the orig_handler already (usually) prints out the panic info - if we don't abort, what's the point of setting the hook?

Reply:
I don't know, but it's this way in develop too https://github.com/holochain/holochain/blob/5806147b4aedcf7aced1b6a88c67ee2ebf941d8c/crates/holochain_lmdb/src/env.rs#L41-L50